include/psa/crypto_values.h

```diff
@@ -2083,10 +2165,38 @@ static inline int mbedtls_svc_key_id_is_null( mbedtls_svc_key_id_t key )
  */
 #define PSA_KEY_USAGE_VERIFY_HASH               ((psa_key_usage_t)0x00002000)
 
-/** Whether the key may be used to derive other keys.
+/** Whether the key may be used to derive other keys or produce a password
+ * hash.
+ *
+ * This flag allows the key to be used as the input of
+ * psa_key_derivation_input_key() at the step
+ * #PSA_KEY_DERIVATION_INPUT_SECRET of #PSA_KEY_DERIVATION_INPUT_PASSWORD
+ * depending on the algorithm, and allows the use of
+ * psa_key_derivation_output_bytes() or psa_key_derivation_output_key()
+ * at the end of the operation.
  */
 #define PSA_KEY_USAGE_DERIVE                    ((psa_key_usage_t)0x00004000)
 
+/** Whether the key may be used to produce a password hash and verify it
+ * against an expected value.
+ *
+ * This flag allows the key to be used as the input of
+ * psa_key_derivation_input_key() at the step
+ * #PSA_KEY_DERIVATION_INPUT_SECRET of #PSA_KEY_DERIVATION_INPUT_PASSWORD
+ * depending on the algorithm, and allows the use of
+ * psa_key_derivation_verify_output_bytes() or
+ * psa_key_derivation_verify_output_key() at the end of the operation.
+ */
+#define PSA_KEY_USAGE_PASSWORD_HASH_AND_VERITY  ((psa_key_usage_t)0x00008000)
```

Comment:
(Using this thread to reply to @athoelke)

> The addition of a permission for a stored key to be used in password verification is also valid. There is no precedent here, as all existing verification APIs take an input buffer parameter. Perhaps password authentication is special?

Yes. For the primary purpose of a password hash, both the password and the hash are confidential. If the hash is leaked, this allows an adversary to conduct offline attacks to brute-force the password. If the adversary can only make queries of the form “is this password correct?”, then the attack has to be online, and so can be rate-limited, logged, and amount-limited. This is why a password hash is stored in a confidential database.

This confidential database may or may not be the PSA key store: depending on the deployments and on where storage is available, I expect that sometimes password hashes will be in the PSA key store (e.g. for a single-user IoT device that allows the user to authenticate with a password) and sometimes they will be in a user database outside the PSA world and PSA will only be used as a crypto implementation (e.g. for a multi-user server using PSA crypto because it has exclusive access to a hardware accelerator).